x/distribution/client/cli/query.go

```diff
@@ -262,3 +302,212 @@ $ %s query distr community-pool
+		if err != nil {
+			return nil, sdkerrors.Wrap(err, fmt.Sprintf("Unable to get block %d from node", previousHeight))
+		}
+
+		currentUnix := currentBlock.BlockMeta.Header.Time.Unix()
+		previousUnix := previousBlock.BlockMeta.Header.Time.Unix()
+
+		if currentUnix >= searchUnix && previousUnix < searchUnix {
+			return &currentHeight, nil
+		} else if currentUnix < searchUnix {
+			minHeight = currentHeight + 1
+		} else {
+			maxHeight = currentHeight - 1
+		}
+	}
+
+	return nil, errors.New("Unable to find block height with search time")
+}
+
+func calculateValidatorRewardsAndCommissionPerDay(
+	node rpcclient.Client,
+	validatorAddress sdk.ValAddress,
+	querier ValidatorQuerier,
+	endDate time.Time,
+	startHeight int64,
+	lastBlockHeight int64) (aggByDay [][]sdk.DecCoins, days []time.Time, error error) {
+	commission := sdk.DecCoins{}
+	reward := sdk.DecCoins{}
+	var lastCommission sdk.DecCoins
+	var lastReward sdk.DecCoins
+	var currentDate time.Time
+	if startHeight > 0 {
+		// We want to start from the previous day's last block, so we can calculate the diff
+		// and account the difference for the starting day's sum.
+		startHeight -= 1
+	}
+	currentHeight := startHeight
+
+	for currentHeight <= lastBlockHeight {
+		currentBlock, err := node.Block(&currentHeight)
+		if err != nil {
+			err = sdkerrors.Wrap(err, "Unable to get block")
+			return
+		}
+
+		if currentBlock.BlockMeta.Header.Time.After(endDate) {
+			break
+		}
+
+		if currentDate.IsZero() {
+			currentDate = currentBlock.BlockMeta.Header.Time
+		}
+
+		if currentBlock.BlockMeta.Header.Time.Day() != currentDate.Day() {
+			// Once we see a block in the next day, record the sum differences
+			// for the previous day
+			aggByDay = append(aggByDay, []sdk.DecCoins{commission, reward})
+			days = append(days, currentDate)
+			currentDate = currentBlock.BlockMeta.Header.Time
+			commission = sdk.DecCoins{}
+			reward = sdk.DecCoins{}
+		}
+
+		validatorCommission, err := querier.GetValidatorCommission(currentHeight, validatorAddress)
+		if err != nil {
+			err = sdkerrors.Wrap(err, "Unable to get validator commission")
+			return
+		}
+
+		if lastCommission != nil {
+			commission = commission.Add(validatorCommission.Sub(lastCommission))
+		}
+		lastCommission = validatorCommission
+
+		validatorRewards, err := querier.GetValidatorRewards(currentHeight, validatorAddress)
+		if lastReward != nil {
+			reward = reward.Add(validatorRewards.Sub(lastReward))
+		}
+		lastReward = validatorRewards
+		currentHeight += 1
+	}
+
+	if !commission.IsZero() || !reward.IsZero() {
+		aggByDay = append(aggByDay, []sdk.DecCoins{commission, reward})
+		days = append(days, currentDate)
+	}
+
+	return
+}
+
+// GetCmdQueryCommunityPool returns the command for fetching community pool info
+func GetCmdQueryGenerateReport(queryRoute string, cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "gen-report [start-date] [end-date] [addr] [addr]...",
```

Comment:
Just a suggestion that aims to slightly improve UX: the positional [end-date] argument may be turned into an optional parameter `--end-date` that, if not supplied, may just fall back to today's date.